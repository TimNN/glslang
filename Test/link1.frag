#version 130

uniform vec4 uv4;
uniform vec3 glass;

const int ci = 8;

vec4 a = ci * uv4;

in vec3 iv3;
in vec4 cup;

void main()
{
}

vec4 b = ci * a;

ivec2 foo(mat2 m)
{
    return ivec2(m[0]);
}

vec4 c = b * b;
