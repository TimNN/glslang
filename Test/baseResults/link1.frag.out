link1.frag

0:? Sequence
0:8  Sequence
0:8    move second child to first child (4-component vector of float)
0:8      'a' (4-component vector of float)
0:8      vector-scale (4-component vector of float)
0:8        8.000000
0:8        'uv4' (uniform 4-component vector of float)
0:13  Function Definition: main( (void)
0:13    Function Parameters: 
0:17  Sequence
0:17    move second child to first child (4-component vector of float)
0:17      'b' (4-component vector of float)
0:17      vector-scale (4-component vector of float)
0:17        8.000000
0:17        'a' (4-component vector of float)
0:19  Function Definition: foo(mf22; (2-component vector of int)
0:19    Function Parameters: 
0:19      'm' (in 2X2 matrix of float)
0:21    Sequence
0:21      Branch: Return with expression
0:21        Convert float to int (2-component vector of int)
0:21          direct index (in 2-component vector of float)
0:21            'm' (in 2X2 matrix of float)
0:21            0 (const int)
0:24  Sequence
0:24    move second child to first child (4-component vector of float)
0:24      'c' (4-component vector of float)
0:24      component-wise multiply (4-component vector of float)
0:24        'b' (4-component vector of float)
0:24        'b' (4-component vector of float)
0:?   Linker Objects
0:?     'uv4' (uniform 4-component vector of float)
0:?     'glass' (uniform 3-component vector of float)
0:?     'iv3' (smooth in 3-component vector of float)
0:?     'cup' (smooth in 4-component vector of float)

link2.frag

0:? Sequence
0:8  Sequence
0:8    move second child to first child (4-component vector of float)
0:8      'd' (4-component vector of float)
0:8      vector-scale (4-component vector of float)
0:8        8.000000
0:8        'uv4' (uniform 4-component vector of float)
0:13  Sequence
0:13    move second child to first child (4-component vector of float)
0:13      'e' (4-component vector of float)
0:13      vector-scale (4-component vector of float)
0:13        8.000000
0:13        'd' (4-component vector of float)
0:15  Function Definition: foo( (2-component vector of int)
0:15    Function Parameters: 
0:17    Sequence
0:17      Branch: Return with expression
0:17        2 (const int)
0:17        2 (const int)
0:20  Sequence
0:20    move second child to first child (4-component vector of float)
0:20      'f' (4-component vector of float)
0:20      component-wise multiply (4-component vector of float)
0:20        'e' (4-component vector of float)
0:20        'e' (4-component vector of float)
0:?   Linker Objects
0:?     'uv4' (uniform 4-component vector of float)
0:?     'glass' (uniform 2-component vector of float)
0:?     'iv3' (smooth in 3-component vector of float)
0:?     'cup' (flat in 4-component vector of float)

link3.frag

0:? Sequence
0:?   Linker Objects
0:?     'iv3' (smooth in highp 2-component vector of float)


Linked fragment stage:

ERROR: Linking fragment stage: Types must match:
    glass: "uniform 3-component vector of float" versus "uniform 2-component vector of float"
ERROR: Linking fragment stage: Interpolation and auxiliary storage qualifiers must match:
    cup: "smooth in 4-component vector of float" versus "flat in 4-component vector of float"
ERROR: Linking fragment stage: Cannot mix ES profile with non-ES profile shaders

ERROR: Linking fragment stage: Types must match:
ERROR: Linking fragment stage: Precision qualifiers must match:
    iv3: "smooth in 3-component vector of float" versus "smooth in highp 2-component vector of float"

0:? Sequence
0:8  Sequence
0:8    move second child to first child (4-component vector of float)
0:8      'a' (4-component vector of float)
0:8      vector-scale (4-component vector of float)
0:8        8.000000
0:8        'uv4' (uniform 4-component vector of float)
0:13  Function Definition: main( (void)
0:13    Function Parameters: 
0:17  Sequence
0:17    move second child to first child (4-component vector of float)
0:17      'b' (4-component vector of float)
0:17      vector-scale (4-component vector of float)
0:17        8.000000
0:17        'a' (4-component vector of float)
0:19  Function Definition: foo(mf22; (2-component vector of int)
0:19    Function Parameters: 
0:19      'm' (in 2X2 matrix of float)
0:21    Sequence
0:21      Branch: Return with expression
0:21        Convert float to int (2-component vector of int)
0:21          direct index (in 2-component vector of float)
0:21            'm' (in 2X2 matrix of float)
0:21            0 (const int)
0:24  Sequence
0:24    move second child to first child (4-component vector of float)
0:24      'c' (4-component vector of float)
0:24      component-wise multiply (4-component vector of float)
0:24        'b' (4-component vector of float)
0:24        'b' (4-component vector of float)
0:8  Sequence
0:8    move second child to first child (4-component vector of float)
0:8      'd' (4-component vector of float)
0:8      vector-scale (4-component vector of float)
0:8        8.000000
0:8        'uv4' (uniform 4-component vector of float)
0:13  Sequence
0:13    move second child to first child (4-component vector of float)
0:13      'e' (4-component vector of float)
0:13      vector-scale (4-component vector of float)
0:13        8.000000
0:13        'd' (4-component vector of float)
0:15  Function Definition: foo( (2-component vector of int)
0:15    Function Parameters: 
0:17    Sequence
0:17      Branch: Return with expression
0:17        2 (const int)
0:17        2 (const int)
0:20  Sequence
0:20    move second child to first child (4-component vector of float)
0:20      'f' (4-component vector of float)
0:20      component-wise multiply (4-component vector of float)
0:20        'e' (4-component vector of float)
0:20        'e' (4-component vector of float)
0:?   Linker Objects
0:?     'uv4' (uniform 4-component vector of float)
0:?     'glass' (uniform 3-component vector of float)
0:?     'iv3' (smooth in 3-component vector of float)
0:?     'cup' (smooth in 4-component vector of float)

